extensions-core/kafka-indexing-service/src/test/java/io/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java

```diff
@@ -2115,12 +2115,8 @@ public void testCheckpointForInactiveTaskGroup()
 
     verifyAll();
 
-    while (serviceEmitter.getStackTrace() != null) {
-      Thread.sleep(100);
-    }
-
-    Assert.assertNull(serviceEmitter.getStackTrace());
-    Assert.assertNull(serviceEmitter.getExceptionMessage());
+    Assert.assertNull(serviceEmitter.getStackTrace(), serviceEmitter.getStackTrace());
```

Comment:
is this a fix or just temp debugging arrangement to see what is in the stacktrace so that later a fix can be done ?

test times out because serviceEmitter.getStackTrace() is non-null , so this assertion is gonna fail in those cases.

Reply:
Yeah, this doesn't fix the root cause, but fixes the infinite loop and print the stack trace so that we can fix it.

Reply:
PR title misled me :)

Reply:
so, do you want this PR to be merged or you're trying builds on this PR to see things whenever this test fails ?

Reply:
Yeah, the title looks somewhat confusing :(. Do you have a better idea? 

I thought it would be great if the CI for this PR shows something interesting, but unfortunately it succeeded. I can restart it until I find something, or check other PRs if they fail with some stack traces. I don't have a strong opinion here, but I would say this PR gives us a small benefit by making CI jobs fail faster.

Reply:
Nice. It catches the exception. I'll check it.

Reply:
@leventov thanks. I've also noticed that before. The problem is that the supervisor periodically executes `runNotice` and this can potentially call every method of the supervisor which in turn requires to mock everything using `EasyMock`. 

I think it's better to refactor the whole KafkaSupervisorTest to be based on a sort of more controllable mockup environment by creating `taskRunner`, `taskMaster`, `taskStorage`, and `taskClient` for test purpose instead of making them using EasyMock.

Raised https://github.com/apache/incubator-druid/issues/6296.